### PR TITLE
[Workers LP] concepts revision

### PR DESCRIPTION
**Pull request by deadlypants1973**
PCX-10073

### CODE REVIEW

File: content/learning-paths/workers/concepts/serverless-computing.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+## On-premise infrastructure
+
+[On-premise infrastructure](https://www.cloudflare.com/learning/cloud/what-is-cloud-migration/) refers to a traditional way of managing computer resources where servers are kept within the physical premises of an organization. The on-premise model gives organizations robust control over the security, customization, and configuration of their infrastructure but has significant downsides such as cost of server maintenance and scaling to accomodate business growth. Cloud computing emerged commercially in the early 2000s as an alternative to this paradigm.
+
+## Cloud infrastructure
+
+[Cloud computing](https://www.cloudflare.com/learning/cloud/what-is-the-cloud/) is defined as hosting computing resources (such as virtual machines, storage, databases, and networking services) on third-party servers. Cloud computing service providers include Amazon Web Services, Microsoft Azure, Gooogle Cloud Platform, and Cloudflare. The mains service models of cloud computing are: Infrastructure-as-a-Service (IaaS), Platform-as-a-Service (Paas), and Software-as-a-Service (SaaS).
```
Comment (penalosa) [inline content/learning-paths/workers/concepts/serverless-computing.md]:
```suggestion
[Cloud computing](https://www.cloudflare.com/learning/cloud/what-is-the-cloud/) is defined as hosting computing resources (such as virtual machines, storage, databases, and networking services) on third-party servers. Cloud computing service providers include Amazon Web Services, Microsoft Azure, Gooogle Cloud Platform, and Cloudflare. The main service models of cloud computing are: Infrastructure-as-a-Service (IaaS), Platform-as-a-Service (Paas), and Software-as-a-Service (SaaS).
```

File: content/learning-paths/workers/concepts/serverless-computing.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+![IaaS, PaaS and Saas models](/images/learning-paths/workers/saas-paas-iaas-diagram.svg)
+
+Cloudflare Workers is under the [Functions-as-a-Service (FaaS)](https://www.cloudflare.com/learning/serverless/glossary/function-as-a-service-faas/) model, which is a serverless way to execute modular pieces of code on the [edge](https://www.cloudflare.com/learning/serverless/glossary/what-is-edge-computing/).
```
Comment (penalosa) [inline content/learning-paths/workers/concepts/serverless-computing.md]: Workers doesn't just support edge computing, with things like smart placement, Durable Objects, and cron triggers all not necessarily running on the edge

File: content/learning-paths/workers/concepts/serverless-computing.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+### Serverless computing
+
+While serverless computing and cloud computing are closely related concepts, there are a few notable differences.
+
+#### Resources management
```
Comment (penalosa) [inline content/learning-paths/workers/concepts/serverless-computing.md]:
```suggestion
#### Resource management
```

File: content/learning-paths/workers/concepts/serverless-computing.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+Cloud computing allows organizations to rent a fixed number of servers or server space. To prepare for seasonal or unplanned spikes in request traffic to their applications, organizations may overpurchase server space to ensure their applications do not go down because of high request volume from end users or customers. In the serverless computing model, organizations and individuals are not required to calculate how much server space or machines they need to rent. Serverless computing providers take care of server management, scaling, and provisioning, allowing developers and organizations to focus on writing and deploying logic.
```
Comment (penalosa) [inline content/learning-paths/workers/concepts/serverless-computing.md]: This disinction isn't really as cut and dry as this. Often cloud computing providers do let you autoscale virtual servers (see https://aws.amazon.com/ec2/autoscaling/, for instance)

File: content/learning-paths/workers/concepts/serverless-computing.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+#### Execution model
+
+Serverless computing differs from traditional cloud computing structures in its cloud architecture. Cloud computing infrastructure includes [virual machines (VM)](https://www.cloudflare.com/learning/cloud/what-is-a-virtual-machine/) and [containers](https://www.cloudflare.com/learning/serverless/serverless-vs-containers/). Serverless computing uses an event-driven model. A serverless application is broken up into functions, and hosted by a serverless computing provider who charges the application developer based on the amount of time each function runs. A function will run when an event (like an HTTP request) occurs. You will learn more about container infrastructure versus Workers infrastructure in the [next part of this module](/learning-paths/workers/concepts/workers-concepts/).
```
Comment (penalosa) [inline content/learning-paths/workers/concepts/serverless-computing.md]: This doesn't really capture the difference between models. Traditional serverless providers (AWS Lambda, for instance) run your functions in a container, and charge based on request duration. Workers are run in V8 isolates, and charge based on CPU time.

File: content/learning-paths/workers/concepts/serverless-computing.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
+---
+title: Serverless computing
```
Comment (penalosa) [inline content/learning-paths/workers/concepts/serverless-computing.md:2]: I'm a bit confused about this page—who's it targeted at? It seems like a general high-level intro to serverless and cloud computing, but it doesn't really talk about Workers and their advantages. Conflating Workers & traditional serverless understates how unique Workers is compared to things like AWS Lambda in things like execution model and billing.

File: content/learning-paths/workers/concepts/serverless-computing.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+#### Execution model
+
+Serverless computing differs from traditional cloud computing structures in its cloud architecture. Cloud computing infrastructure includes [virual machines (VM)](https://www.cloudflare.com/learning/cloud/what-is-a-virtual-machine/) and [containers](https://www.cloudflare.com/learning/serverless/serverless-vs-containers/). Serverless computing uses an event-driven model. A serverless application is broken up into functions, and hosted by a serverless computing provider who charges the application developer based on the amount of time each function runs. A function will run when an event (like an HTTP request) occurs. You will learn more about container infrastructure versus Workers infrastructure in the [next part of this module](/learning-paths/workers/concepts/workers-concepts/). 
+
+#### Billing model
+
+Developers and organizations using serverless computing are billed on a [usage model](/workers/pricing/) paradigm. Instead of paying for a fixed amount of computing resources that may be underutilized or exceeded, users pay as much as they use in the serverless model.
```
Comment (penalosa) [inline content/learning-paths/workers/concepts/serverless-computing.md]: "Usage model" can mean many things, and isn't generalisable across serverless computing platforms. Lambda and Workers are both serverless computing platforms, for instance, but charge based on usage very differently.

File: content/learning-paths/workers/concepts/what-is-cloudflare.md [hunk 8]
```diff
@@ -0,0 +1,64 @@
+---
+title: What is Cloudflare?
```
Comment (penalosa) [inline content/learning-paths/workers/concepts/what-is-cloudflare.md]: This should link to https://www.cloudflare.com/learning/what-is-cloudflare/

File: content/learning-paths/workers/concepts/workers-concepts.md [hunk 9]
```diff
@@ -1,36 +1,53 @@
...
-title: Workers concepts
+title: Cloudflare Workers
 pcx_content_type: learning-unit
-weight: 1
+weight: 3
 layout: learning-unit
 ---
 
-## Workers concepts
+Cloudflare Workers has significant differences in its architecture from other serverless computing providers. The Workers runtime is designed is designed to be JavaScript-standards compliant and web-interoperable. It uses the V8 engine — the same engine used by Chromium and Node.js and has an open-source version, [`workerd`](https://github.com/cloudflare/workerd).
```
Comment (penalosa) [inline content/learning-paths/workers/concepts/workers-concepts.md]:
```suggestion
Cloudflare Workers has significant differences in its architecture from other serverless computing providers. The Workers runtime is designed to be JavaScript-standards compliant and web-interoperable. It uses the V8 engine — the same engine used by Chromium and Node.js, and has an open-source version, [`workerd`](https://github.com/cloudflare/workerd).
```

File: content/learning-paths/workers/concepts/workers-concepts.md [hunk 10]
```diff
@@ -1,36 +1,53 @@
...
 
-Cloudflare Workers allow you to build web functions and applications without configuring or maintaining infrastructure. Workers are deployed globally to over 300+ data centers around the world on Cloudflare’s global network.
+## Workers
 
-Before you begin building with Workers, review the following reference material to understand how Workers works.
+Workers projects are made of two basic components:
 
-### Serverless computing
+1. The entry point (this is the initial `index.js` or `index.ts` that is generated in project creation) where you write code.
+2. The [`wrangler.toml` file](/workers/wrangler/configuration/) where you define your project's configuration.
```
Comment (penalosa) [inline content/learning-paths/workers/concepts/workers-concepts.md]: This is potentially _correct_, but I'm not sure it's the right way to phrase it. Workers consist of multiple modules (JS/WASM/Text), along with an optional `wrangler.toml` file depending on whether you're using Wrangler or not.

File: content/learning-paths/workers/concepts/workers-concepts.md [hunk 11]
```diff
@@ -1,36 +1,53 @@
...
+If you are building your Worker entirely on the Cloudflare dashboard, you will use the dashboard editor to update your code and the Cloudflare dashboard configuration settings to configure your project.
 
-### Workers
+When an event (such as an HTTP request or a [Cron Trigger](/workers/configuration/cron-triggers/)) invokes a Worker, the Worker code will execute. Workers projects can have complex functionality, run on multiple [routes](/workers/configuration/routing/), and route function logic according to its intended destination. Cloudflare Workers can do a lot. A single Worker project can have logic as complex or as simple as the developer desires. A simple use case might be [Returning a small HTML page](/workers/examples/return-html/) while a more complex project would span multiple domains, multiple routes for each domain, and different logic for each route and domain. The architectural complexity of a Worker project is decided by the developer.
```
Comment (penalosa) [inline content/learning-paths/workers/concepts/workers-concepts.md]: This paragraph is great, but it feels a bit messy, and like it's re-stating the key idea (workers can do anything) multiple times

File: content/learning-paths/workers/concepts/workers-concepts.md [hunk 12]
```diff
@@ -1,36 +1,53 @@
...
 
-Explore Cloudflare Workers developer documentation by starting with [How Workers works](/workers/reference/how-workers-works/), and learn the difference between the Workers runtime versus traditional browsers and Node.js. If you are a Node.js user, review [Node.js compatibility](/workers/runtime-apis/nodejs/) in the Workers documentation.
+## Execution
 
-### The cache
+Workers run on [Cloudflare's global network](https://www.cloudflare.com/network/) - a growing global network of thousands of machines distributed across hundreds of locations. Only one Workers instance runs on each of the many Cloudflare global network servers. An instance refers to a single execution environment where Cloudflare Workers runs. Every Worker running within an instance is executed within its own isolate. Workers are stateless, meaning that they do not maintain data or state between invocations.
```
Comment (penalosa) [inline content/learning-paths/workers/concepts/workers-concepts.md]: I feel like "growing" implies that it's currently small. Why did we remove the 300+ marketing?

Comment (penalosa) [reply content/learning-paths/workers/concepts/workers-concepts.md]: What's an "instance"? I don't see it referenced anywhere else on the page

File: content/workers/_partials/_compute-per-request.md [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+addEventListener('fetch', event => {
+  event.respondWith(handleRequest(event.request));
+});
+
+async function handleRequest(request) {
+  return new Response('Hello worker!', { status: 200 });
+}
+```
+{{</tab>}}
+{{</tabs>}}
```
Comment (penalosa) [inline content/workers/_partials/_compute-per-request.md]: We shouldn't document service workers here—this should have a tab for JS and a tab for TS

**Review by penalosa — CHANGES_REQUESTED** [review]

File: content/learning-paths/workers/concepts/what-is-cloudflare.md (hunk 8 above)
Comment (deadlypants1973) [reply content/learning-paths/workers/concepts/what-is-cloudflare.md]: Titles don't get links. This link is highlighted in an aside further on the page.

**Review by penalosa — APPROVED** [review]

**Review by kodster28 — APPROVED** [review]

**Pull request merged**